packages/legend-manual-tests/src/__tests__/query-execution/TEMPORARY__QueryBuilderExecution.test.tsx

```diff
@@ -183,4 +183,78 @@ test(integrationTest('test query execution with parameters'), async () => {
   expect(
     queryBuilderResultPanel.getElementsByClassName('ag-cell')[0]?.innerHTML,
   ).toContain('20');
+
+  // Test drag and drop a new property to Projection panel and ag-grid is updated after re-running query
+  const queryBuilderExploreTree = await waitFor(() =>
+    renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_EXPLORER),
+  );
+  const propertyNode = getByText(queryBuilderExploreTree, 'First Name');
+  await waitFor(() => fireEvent.contextMenu(propertyNode));
+  await waitFor(() =>
+    fireEvent.click(renderResult.getByText('Add Property to Fetch Structure')),
+  );
+  const queryBuilderResultPanel1 = await waitFor(() =>
+    renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_RESULT_PANEL),
+  );
+  await waitFor(() =>
+    findByText(queryBuilderResultPanel1, 'Preview data might be stale'),
+  );
+  const executionInput1 = (
+    queryBuilderState.graphManagerState.graphManager as V1_PureGraphManager
+  ).createExecutionInput(
+    queryBuilderState.graphManagerState.graph,
+    guaranteeNonNullable(queryBuilderState.executionContextState.mapping),
+    queryBuilderState.resultState.buildExecutionRawLambda(),
+    guaranteeNonNullable(queryBuilderState.executionContextState.runtimeValue),
+    V1_PureGraphManager.DEV_PROTOCOL_VERSION,
+    parameterValues,
+  );
+  const executionResult1 = await axios.post<
+    unknown,
+    AxiosResponse<{ elements: object[] }>
+  >(
+    `${ENGINE_SERVER_URL}/pure/v1/execution/execute`,
+    V1_ExecuteInput.serialization.toJson(executionInput1),
+    {
+      headers: {
+        [HttpHeader.CONTENT_TYPE]: ContentType.APPLICATION_JSON,
+      },
+    },
+  );
+  createSpy(
+    queryBuilderState.graphManagerState.graphManager,
+    'runQuery',
+  ).mockResolvedValue(
+    V1_buildExecutionResult(V1_serializeExecutionResult(executionResult1.data)),
+  );
+  await act(async () => {
+    fireEvent.click(getByText(queryBuilderResultPanel1, 'Run Query'));
+  });
+  const parameterValueDialog1 = await waitFor(() =>
+    renderResult.getByRole('dialog'),
+  );
+  await waitFor(() =>
+    fireEvent.change(
+      guaranteeNonNullable(
+        parameterValueDialog1.getElementsByClassName(
+          'value-spec-editor__input',
+        )[0],
+      ),
+      {
+        target: { value: 20 },
+      },
+    ),
+  );
+  await waitFor(() => fireEvent.click(getByText(parameterValueDialog1, 'Run')));
+
+  expect(
+    queryBuilderResultPanel1.getElementsByClassName(
```

Comment:
same thing here, could we query maybe by testId? add a testID on the result panel?